### DISCUSSION TITLE
how to update `sw.js` in production?

### DISCUSSION THREAD

**Discussion by jamaluddinrumi** [vite-pwa/vite-plugin-pwa#768] [Q&A] [answered]
i had some mistakes for my prior auto-generated `sw.js` by vite-pwa but unfortunately it's already been deployed.

i need to update the production version of `sw.js` as soon as possible, but the cdn i used is always returning the old one.

i need a possibility to set `updateViaCache: all` option when registering service worker.

```typescript
if ('serviceWorker' in navigator) {
  navigator.serviceWorker.register('/service-worker.js', {
    updateViaCache: 'all',
  });
}
```

i've already take a look on vite-pwa `src/client/build/register.ts`
```typescript
      // register the service worker
      wb.register({ immediate }).then((r) => {
        if (onRegisteredSW)
          onRegisteredSW('__SW__', r)
        else
          onRegistered?.(r)
      }).catch((e) => {
        onRegisterError?.(e)
      })
```

but, its already registered with `immediate` option only.

where should i put `updateViaCache: all` option?

references: https://developer.chrome.com/blog/fresher-sw/#updateviacache

**Comment by jamaluddinrumi** [comment]
i'm just read https://developer.chrome.com/docs/workbox/modules/workbox-window#when_an_updated_version_of_the_service_worker_is_found

and, i think i need to [manually triggering update checks](https://developer.chrome.com/docs/workbox/service-worker-lifecycle#manually_triggering_update_checks)

```typescript
navigator.serviceWorker.ready.then((registration) => {
  registration.update();
});
```

but, where should i put it to?

**Comment by jamaluddinrumi** [comment]
i use 

```.env
Astro version            v2.10.15
Package manager          pnpm
Platform                 darwin
Architecture             arm64
Adapter                  Couldn't determine.
Integrations             astro-i18next, astro-imagetools, @storyblok/astro, @astrojs/tailwind, @astrojs/sitemap, astro-robots-txt, @astrojs/partytown, @vite-pwa/astro-integration, astro-compress, astro-critters
```

and now, i end up put it to my `src/pwa.ts` 
```typescript
import { registerSW } from "virtual:pwa-register"

registerSW({
  immediate: true,
  onNeedRefresh: () => {
    navigator.serviceWorker.ready.then((registration) => { 
      registration.update()
    })
  }
})
```

reference: https://vite-pwa-org.netlify.app/frameworks/astro.html#auto-update

any helps very appreciated 🙏

**Reply by userquin** [reply, marked as answer]
You can destroy your service worker and include a new one: check this guide https://vite-pwa-org.netlify.app/guide/unregister-service-worker.html

Read the page and ask if you need help, ping me on discord server here (pwa channel): https://discord.gg/BFrgqvVV

**Reply by jamaluddinrumi** [reply]
for now, i'll watch my setup above, is it going to works? if, not i'll try [unregister service worker](https://vite-pwa-org.netlify.app/guide/unregister-service-worker.html#unregister-service-worker).

and, already join the discord server 🙏
